### PR TITLE
[v6r16] Create sudoComputingElement.py

### PR DESCRIPTION
**Pull request by Andrew-McNab-UK**
This PR adds the sudoComputingElement used within VMs to run each payload as a different user. Otherwise, it operates the same as the InProcess CE.

### CODE REVIEW

File: Resources/Computing/sudoComputingElement.py [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+      self.log.error( 'X509_USER_PROXY variable for pilot proxy not found in local environment' )
+      return S_ERROR( 'X509_USER_PROXY not found' )
+
+    pilotProxy = os.environ['X509_USER_PROXY']
+    self.log.info( 'Pilot proxy X509_USER_PROXY=%s' % pilotProxy )
+
+    # First username in the sequence to use when running payload job
+    # If first is pltXXp00 then have pltXXp01, pltXXp02, ...
+    try:
+      baseUsername = gConfig.getValue( "/LocalSite/SudoBaseUsername", "" )
```
Comment (atsareg) [inline Resources/Computing/sudoComputingElement.py]: ComputingElement configuration should be taken from the usual place as for any CEs and obtained in the code from the ceParameters variable

**Comment by atsareg** [comment]
Why do you need new payload user name for each payload ? This computing element does not execute several payloads in parallel. So, if all the remnants of the previous payload are properly cleaned, subsequent payload can run with the same payload user without a risk to spy on the previous payload sensitive information.

**Comment by Andrew-McNab-UK** [comment]
@atsareg The tricky word in your comment is "if". Reusing unix users for different people is a bad idea when you can avoid it. When running inside your own VM, it's cheap to pre-create plenty of payload user accounts. This means cleaning up between payloads is just about reclaiming space, rather than trying to ensure traceability and isolation. 
I've updated the PR to use PayloadUser from ceParameters as you suggested by email if set, and otherwise to look for BaseUsername from ceParameters rather than with gConfig and a path. I think the PR as it now stands will work in the scenario you have in mind and what I need for VMs using the new Pilots.

**Comment by fstagni** [comment]
Hi, you made this PR targeting integration branch. I changed it to target rel-v6r16 (as it should) and it added several un-related commits. Please, Andrew, later re-do it targeting rel-v6r16.

**Comment by Andrew-McNab-UK** [comment]
Ok.

**Pull request closed without merging**